lib/puppet_x/bsd/hostname_if/vlan.rb

```diff
@@ -54,6 +54,9 @@ def content
 
         def vlan_string
           vlanstring = []
+          if @config[:id].to_i < 1 or @config[:id].to_i > 4094
```

Comment:
It's part of the 802.1q specification, which says that the VLAN ID is a 12 bit field, with 0x0 and 0xFFF reserved, see: https://en.wikipedia.org/wiki/IEEE_802.1Q